Source/Workers/transcodeCRNToDXT.js

```diff
@@ -0,0 +1,134 @@
+    RuntimeError,
+    crunch,
+    createTaskProcessorWorker
+) {
+    'use strict';
+
+    // Modified from: https://github.com/toji/texture-tester/blob/master/js/webgl-texture-util.js
+    // Also see: http://toji.github.io/texture-tester/
+
+    /* Copyright (c) 2014, Brandon Jones. All rights reserved.
+
+     Redistribution and use in source and binary forms, with or without modification,
+     are permitted provided that the following conditions are met:
+
+     * Redistributions of source code must retain the above copyright notice, this
+     list of conditions and the following disclaimer.
+     * Redistributions in binary form must reproduce the above copyright notice,
+     this list of conditions and the following disclaimer in the documentation
+     and/or other materials provided with the distribution.
+
+     THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+     ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+     WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+     DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+     ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+     (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+     LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+     ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+     (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+     SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. */
+
+    // Taken from crnlib.h
+    var CRN_FORMAT = {
+        cCRNFmtInvalid: -1,
+
+        cCRNFmtDXT1: 0,
+        // cCRNFmtDXT3 is not currently supported when writing to CRN - only DDS.
+        cCRNFmtDXT3: 1,
+        cCRNFmtDXT5: 2
+
+        // Crunch supports more formats than this, but we can't use them here.
+    };
+
+    // Mapping of Crunch formats to DXT formats.
+    var DXT_FORMAT_MAP = {};
+    DXT_FORMAT_MAP[CRN_FORMAT.cCRNFmtDXT1] = PixelFormat.RGB_DXT1;
+    DXT_FORMAT_MAP[CRN_FORMAT.cCRNFmtDXT3] = PixelFormat.RGBA_DXT3;
+    DXT_FORMAT_MAP[CRN_FORMAT.cCRNFmtDXT5] = PixelFormat.RGBA_DXT5;
+
+
+    var dst;
+    var dxtData;
+    var cachedDstSize = 0;
+
+    // Copy an array of bytes into or out of the emscripten heap.
+    function arrayBufferCopy(src, dst, dstByteOffset, numBytes) {
+        var i;
+        var dst32Offset = dstByteOffset / 4;
+        var tail = (numBytes % 4);
+        var src32 = new Uint32Array(src.buffer, 0, (numBytes - tail) / 4);
+        var dst32 = new Uint32Array(dst.buffer);
+        for (i = 0; i < src32.length; i++) {
+            dst32[dst32Offset + i] = src32[i];
+        }
+        for (i = numBytes - tail; i < numBytes; i++) {
+            dst[dstByteOffset + i] = src[i];
+        }
+    }
+
+    function transcodeCRNToDXT(arrayBuffer, transferableObjects) {
```

Comment:
Not sure how we document workers, but I would explicitly mark this `@private` as we do elsewhere.